src/lat/word-align-lattice-lexicon.cc

```diff
@@ -1038,8 +1038,18 @@ bool WordAlignLatticeLexicon(const CompactLattice &lat,
   // We'll let the calling code warn if this is false; it will know the utterance-id.
   ans = aligner.AlignLattice() && ans;
   if (ans && opts.test) { // We only test if it succeeded.
-    if (!TestWordAlignedLattice(lexicon_info, tmodel, lat, *lat_out,
-                                opts.allow_duplicate_paths)) {
+    const TransitionModel* transition_model;
+    if ((transition_model = dynamic_cast<const TransitionModel*>(&tmodel)) == nullptr) {
+        KALDI_WARN << "Lattice test via TestWordAlignedLattice does not support "
+                   << "non-TransitionModel TransitionInformation instances. No "
+                   << "testing will be done.";
+    }
+    // this function brings in Posterior and TransitionModel from
+    // "hmm/transition-model.h" and "hmm/hmm-utils.h". It would be
+    // nice to be able to fully remove those dependencies from this
+    // function, given that it is only for sanity checking.
+    else if (!TestWordAlignedLattice(lexicon_info, *transition_model, lat, *lat_out,
+                                     opts.allow_duplicate_paths)) {
```

Comment:
This really comes out too ugly. Just remove the condition check and the call to `TestWordAlignedLattice` into latbin/lattice-align-words-lexicon.cc, the only place it's called from, and the function `TestWordAlignedLattice` itself into the TransitionModel-dependent lattice-functions .cc/.h file.